runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -1186,10 +1186,23 @@ class INTERPRETER_CLASS
 					_currentThread->tempSlot = (UDATA)invalidIndex;
 					rc = THROW_AIOB;
 				} else {
-					I_32 value = VM_ArrayCopyHelpers::referenceArrayCopy(_currentThread, srcObject, srcStart, destObject, destStart, elementCount);
-					if (-1 != value) {
-						buildInternalNativeStackFrame(REGISTER_ARGS);
-						rc = THROW_ARRAY_STORE;
+					if (elementCount > 0) {
+						J9Class *srcComponentClass = ((J9ArrayClass *)J9OBJECT_CLAZZ(_currentThread, srcObject))->componentType;
+						J9Class *destComponentClass = ((J9ArrayClass *)J9OBJECT_CLAZZ(_currentThread, destObject))->componentType;
+
+						if (J9_IS_J9CLASS_PRIMITIVE_VALUETYPE(destComponentClass) &&
+							!J9_IS_J9CLASS_PRIMITIVE_VALUETYPE(srcComponentClass) &&
```

Comment:
Do we need to check if `srcComponentClass` is primitive VT ? It could be an array of Object and `srcComponentClass` is `java.lang.Object`.

Reply:
Yeah I suppose that check is unnecessary since I'm already checking if the problem element is null on the next line

Reply:
removed in [92cb616](https://github.com/eclipse-openj9/openj9/commit/92cb6169f6ef83030ce3d192917cf7b963164ba2)